### ISSUE TITLE
Accuracy for contrastive loss layer.

### ISSUE THREAD

**Issue by cheer37** [BVLC/caffe#3809]
I am training the siamese network using contrastive loss layer.
But i see only the loss, so i dont know my net going well or not, and also how much.
I want the feature of computing the accuracy for contrastive loss layer in caffe.
Thanks.

**Comment by jenifferYingyiWu** [comment]
Hi,
    After the images in train.txt haven been trained, and the images in val.txt have been tested.
How to calculate the "precision" and "recall" from the val.txt file (which has been tested by Caffe model)?

   Could you pls help to recommend some python codes?

Thanks!

**Comment by seanbell** [comment]
@cheer37 please ask these kinds of questions on the mailing list.  Github Issues are for discussing the code itself, architecture changes, bugs.  It's not for asking questions about whether or not caffe supports a feature or how to do things.

https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.

**Comment by seanbell** [comment]
I agree that it would be helpful to add this, where accuracy is defined with respect to the contrastive loss threshold.  Feel free to submit a PR implementing this feature!
